include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -279,6 +279,12 @@ class binary_writer
 
             case value_t::binary:
             {
+                if (j.m_value.binary->has_subtype())
+                {
+                    write_number(static_cast<std::uint8_t>(0xd8));
```

Comment:
What does `0xd8` mean?

Reply:
It's a single byte data tag.

> 0xd8..0xdb : more tagged items, 1/2/4/8 bytes and then a data item follow

from https://tools.ietf.org/html/rfc7049

if we want to support bigger than 1 byte, it would be 0xd9, 0xda, 0xdb instead.